crates/core/src/parser/parser.rs

```diff
@@ -78,6 +78,16 @@ pub fn parse(text: &str, options: &Language) -> Result<Vec<Box<dyn Renderable>>>
     Ok(renderables)
 }
 
+/// Given a `Variable` as a string, parses it into a `Variable`.
+pub fn parse_variable_from_text(text: &str) -> Result<Variable> {
+    let variable = LiquidParser::parse(Rule::Variable, text)
+        .map_err(convert_pest_error)?
+        .next()
+        .expect("Parsing a variable failed.");
+
+    Ok(parse_variable(variable))
+}
+
```

Comment:
Understand if you're uncomfortable introducing a new public function here. If you take issue with this, what would you recommend?

Reply:
We need to re-do the parsing which will be a breaking change anyways, its likely fine to expose this for now